pkg/aws/core.go

```diff
@@ -565,37 +565,35 @@ func (d *Driver) ListMachines(ctx context.Context, req *driver.ListMachinesReque
 	}
 
 	listOfVMs := make(map[string]string)
-	var nextToken *string
+	paginator := ec2.NewDescribeInstancesPaginator(client, input, func(opt *ec2.DescribeInstancesPaginatorOptions) {
+		opt.StopOnDuplicateToken = true
+	})
 	pageCount := 0
 
-	for {
-		input.NextToken = nextToken
-
-		runResult, err := client.DescribeInstances(ctx, input)
+	for paginator.HasMorePages() {
+		pageCount++
+		klog.V(3).Infof("Fetching page %d of ListMachines for machineClass %q", pageCount, machineClass.Name)
```

Comment:
Since machineClass.Name is a string, %s should be sufficient. Unless your intention was to have a quoted value
```suggestion
		klog.V(3).Infof("Fetching page %d of ListMachines for machineClass %s", pageCount, machineClass.Name)
```